S3/ConnMan.py

```diff
@@ -28,11 +29,27 @@ class http_connection(object):
 
     @staticmethod
     def _ssl_verified_context(cafile):
+        cfg = Config()
         context = None
         try:
             context = ssl.create_default_context(cafile=cafile)
         except AttributeError: # no ssl.create_default_context
             pass
+        if context and not cfg.check_ssl_hostname:
+            context.check_hostname = False
+            debug(u'Disabling SSL certificate hostname checking')
+
+        return context
+
+    @staticmethod
+    def _ssl_unverified_context(cafile):
+        debug(u'Disabling SSL certificate checking')
+        context = None
+        try:
+            context = ssl._create_unverified_context(cafile=cafile,
+                                                     cert_reqs=CERT_NONE)
+        except AttributeError: # no ssl._create_unverified_context
+            pass
```

Comment:
I'm not expert with python and ssl, but why using the "_create_unverified_context()" function, instead of simply using "context.verify_mode = CERT_NONE" after the  ssl.create_default_context(cafile=cafile) ?
If that makes sense, all of that could be factorized in one function i guess.